python/pyspark/sql/tests/connect/test_connect_creation.py

```diff
@@ -554,6 +554,31 @@ def test_create_dataframe_from_pandas_with_ns_timestamp(self):
                 self.spark.createDataFrame(pdf).collect(),
             )
 
+    def test_schema_inference_from_pandas_with_dict(self):
+        from pyspark.sql.connect import functions as CF
+        from pyspark.sql import functions as F
+
+        pdf = pd.DataFrame({"str_col": ["second"], "dict_col": [{"first": 0.7, "second": 0.3}]})
+
+        self.connect.conf.set("spark.sql.execution.pandas.inferPandasDictAsMap", True)
```

Comment:
can we use `with self.sql_conf`? Otherwise, this will affect the other test cases as well.

Reply:
I tried testing `with self.sql_conf` but it seems not working for some reason. Maybe we should define a config for `connector/connect/server/src/main/scala/org/apache/spark/sql/connect/config/Connect.scala`??

I fixed test to manually reset the conf for now.